pkg/router/controller/factory/factory.go

```diff
@@ -18,17 +18,48 @@ import (
 	"github.com/openshift/origin/pkg/router/controller"
 )
 
+// RouterControllerFactory initializes and manages the watches that drive a router
+// controller. It supports optional scoping on Namespace, Labels, and Fields of routes.
+// If Namespace is empty, it means "all namespaces".
 type RouterControllerFactory struct {
-	KClient  kclient.Interface
-	OSClient osclient.Interface
+	KClient        kclient.EndpointsNamespacer
+	OSClient       osclient.RoutesNamespacer
+	ResyncInterval time.Duration
+	Namespace      string
+	Labels         labels.Selector
+	Fields         fields.Selector
 }
 
+// NewDefaultRouterControllerFactory initializes a default router controller factory.
+func NewDefaultRouterControllerFactory(oc osclient.RoutesNamespacer, kc kclient.EndpointsNamespacer) *RouterControllerFactory {
+	return &RouterControllerFactory{
+		KClient:        kc,
+		OSClient:       oc,
+		ResyncInterval: 10 * time.Minute,
```

Comment:
I'll make it a flag - resync of 100k routes every few minutes would be too
often.

On Wed, Sep 2, 2015 at 2:40 PM, Ram Ranganathan <notifications@github.com>
wrote:

> In pkg/router/controller/factory/factory.go
> <https://github.com/openshift/origin/pull/4421#discussion_r38568384>:
>
> >  }
> >
> > +// NewDefaultRouterControllerFactory initializes a default router controller factory.
> > +func NewDefaultRouterControllerFactory(oc osclient.RoutesNamespacer, kc kclient.EndpointsNamespacer) *RouterControllerFactory {
> > +	return &RouterControllerFactory{
> > +		KClient:        kc,
> > +		OSClient:       oc,
> > +		ResyncInterval: 10 * time.Minute,
>
> Not that I mind (gives me more time to muck around with the haproxy config
> and test things inside the container) but the earlier one was a couple of
> minutes.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/4421/files#r38568384>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift